cmd/packageloader/loader/loader_test.go

```diff
@@ -0,0 +1,163 @@
+package loader
```

Comment:
The test should be `package loader_test`

Reply:
Why should it be a different package? It would make it impossible to do structural tests, unless I misunderstood and we should only be doing functional tests

Reply:
We can export private functions, create export_test or something then inside it just export the private functions for testing. `var SamplePrivateFunction=samplePrivateFunction`

Reply:
Or have separate unit test file for private functions and that can be `package loader`

Reply:
Alright, I will take the first approach since almost all functions are private